front/src/applications/operationalStudies/components/MacroEditor/types.ts

```diff
@@ -154,8 +154,9 @@ export type NetzgrafikDto = {
   };
 };
 
-export type NGETrainrunEvent = {
+export type NGEEvent = {
   type: 'create' | 'delete' | 'update';
-  objectType: 'trainrun';
-  trainrun: Trainrun;
+  objectType: 'node' | 'trainrun';
+  trainrun?: Trainrun;
+  node?: Node;
```

Comment:
Use a type like this, this way we know in the type that if the objectType is node, then we have a Node and same for the trainrun
```suggestion
  } & (
  | {
      objectType: 'trainrun';
      trainrun: Trainrun;
    }
  | { objectType: 'node'; node: Node }
);
```